lib/class-wp-block.php

```diff
@@ -0,0 +1,234 @@
+<?php
+/**
+ * Blocks API: WP_Block class
+ *
+ * @package Gutenberg
+ */
+
+/**
+ * Class representing a parsed instance of a block.
+ */
+class WP_Block implements ArrayAccess {
+
+	/**
+	 * Name of block.
+	 *
+	 * @example "core/paragraph"
+	 *
+	 * @var string
+	 */
+	public $name;
+
+	/**
+	 * Block type associated with the instance.
+	 *
+	 * @var WP_Block_Type
+	 */
+	public $block_type;
+
+	/**
+	 * Block context values.
+	 *
+	 * @var array
+	 */
+	public $context = [];
```

Comment:
> The PHP Coding Standards requires `array()` to be used instead of `[]`.
> 
> https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#declaring-arrays

Hm.

Hmmmmmmm......

Okay :slightly_smiling_face:

Reply:
>(I wonder why the linter didn't complain?)

It's configured to allow any syntax available in 5.6:

https://github.com/WordPress/gutenberg/blob/0336a3417024537a58cab3114cfbb12a5adabc83/phpcs.xml.dist#L6

My guess: We previously relied on this to have been caught when it was configured as 5.3, and when we updated to 5.6, there were no rules which explicitly forbid the short syntax, so they're considered "valid".

It's useful for me to temporarily revert this to 5.3 to find all occurrences, but we want to support other 5.6 features, so it's not a permanent change we'd want to make.

My guess is we'll need to either:

- Await some decision whether short syntax will be reconsidered.
   - ["Short array syntax will be revisited at a later time"](https://make.wordpress.org/core/2020/03/20/updating-the-coding-standards-for-modern-php/)
- Try to find some appropriate configuration to forbid this syntax in the meantime, if any exist.

Reply:
> * Try to find some appropriate configuration to forbid this syntax in the meantime, if any exist.

#21677

Reply:
> The PHP Coding Standards requires `array()` to be used instead of `[]`.

Updated in ad677ec.